### PR TITLE
Allow arbitrary keys for normal_ordering

### PR DESCRIPTION
**Pull request by idk3**

### CODE REVIEW

**Comment by babbush** [comment]
As we discussed in person, I think there is a better solution to the problem this solves so I will close the pull request for now.

File: src/fermilib/ops/_fermion_operator.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def number_operator(n_orbitals, orbital=None, coefficient=1.):
     return operator
 
 
-def normal_ordered_term(term, coefficient):
+def normal_ordered_term(term, coefficient, key_array=None):
```
Comment (Spaceenter) [inline src/fermilib/ops/_fermion_operator.py]: Add doc for "key_array" in the 'Args:' below?

Comment (idk3) [inline src/fermilib/ops/_fermion_operator.py]: I don't think Ryan will allow the pull request either way, but key_array gives the assignment from range(n) to an arbitrary assignment. For example if you normal order 3^ 0^ by [0, 1, 2, 3] it'll leave it as is, but if you normal order by [3, 2, 1, 0] it'll reverse it to -0^ 3^. There are a lot of other examples in the test file for this pull request.

**Comment by babbush** [comment]
As Damian mentioned, I should really explain why we made certain decisions when discussed offline so people know what is going on. Basically, there is currently a mismatch in convention between code that Ian is writing to apply a Fourier transformation to a state and the code that Wei wrote to apply the Fourier to mode operators. Ian figured out he could fix this by enforcing a different normal ordering after using Wei's code. But I think it is confusing and unnecessary to have two different normal ordering procedures. I told Ian it was better to change Wei's code so that it conforms to Ian's code. That way all of our code would be compatible with the same normal ordering convention.

**Pull request closed without merging**